misc/i18n/generateTranslation.js

```diff
@@ -0,0 +1,412 @@
+import { writeFile } from 'fs'
+import AvatarConfig from './AvatarConfig.json' assert {type: 'json'}
+import skillConfig from './AvatarSkillConfig.json' assert {type: 'json'}
+import lightconeConfig from './EquipmentConfig.json' assert {type: 'json'}
+import lightconeRankConfig from './EquipmentSkillConfig.json' assert {type: 'json'}
+import relicsetConfig from './RelicSetConfig.json' assert {type: 'json'}
+import relicEffectConfig from './RelicSetSkillConfig.json' assert {type: 'json'}
+import damageConfig from './DamageType.json' assert {type: 'json'}
+import pathConfig from './AvatarBaseType.json' assert {type: 'json'}
+import rankConfig from './AvatarRankConfig.json' assert {type: 'json'}
+import statusConfig from './AvatarStatusConfig.json' assert {type: 'json'}
+import traceConfig from './AvatarSkillTreeConfig.json' assert {type: 'json'}
+import TextMapZH from './TextMapCHS.json' assert {type: 'json'}
+import TextMapDE from './TextMapDE.json' assert {type: 'json'}
+import TextMapEN from './TextMapEN.json' assert {type: 'json'}
+import TextMapES from './TextMapES.json' assert {type: 'json'}
+import TextMapFR from './TextMapFR.json' assert {type: 'json'}
+import TextMapID from './TextMapID.json' assert {type: 'json'}
+import TextMapJP from './TextMapJP.json' assert {type: 'json'}
+import TextMapKR from './TextMapKR.json' assert {type: 'json'}
+import TextMapPT from './TextMapPT.json' assert {type: 'json'}
+import TextMapRU from './TextMapRU.json' assert {type: 'json'}
+import TextMapTH from './TextMapTH.json' assert {type: 'json'}
+import TextMapVI from './TextMapVI.json' assert {type: 'json'}
+
+function precisionRound(number) {
```

Comment:
Can we TS this file and put some info on how to run it? I think the ideal end state is to be able to run the script with `npx tsx misc/i18n/generateTranslations.ts`

But I'm getting errors with that

```
import AvatarConfig from './AvatarConfig.json' assert {type: 'json'}
                                               ^^^^^^

SyntaxError: Unexpected identifier 'assert'
    at compileSourceTextModule (node:internal/modules/esm/utils:337:16)
    at ModuleLoader.moduleStrategy (node:internal/modules/esm/translators:166:18)
    at callTranslator (node:internal/modules/esm/loader:436:14)
    at ModuleLoader.moduleProvider (node:internal/modules/esm/loader:442:30)
    at async ModuleJob._link (node:internal/modules/esm/module_job:106:19)

Node.js v22.6.0
```